autoscale-workload-rabbit/src/main/java/com/github/autoscaler/workload/rabbit/RabbitSystemResourceMonitor.java

```diff
@@ -109,9 +109,14 @@ private Optional<Integer> getOffloadingDiskFreeMb() throws ScalerException
         if (offloadingDir == null || offloadingDir.isEmpty()) {
             throw new ScalerException("Payload offloading directory is not configured.");
         }
+
+        if (!Files.exists(Paths.get(offloadingDir))) {
```

Comment:
If you extract `offloadingPath` then you can simplify the `Files.getFileStore()` line a few lines further down.

```suggestion
        final Path offloadingPath = Paths.get(offloadingDir);
        if (!Files.exists(offloadingPath)) {
```